doc/tutorials/snippet-sharing-rest-service.md

```diff
@@ -0,0 +1,196 @@
+- Boost >= v1.45
+- OpenSSL
+- LibJWT >= v1.5
+    - Jansson
+    - OpenSSL
+
+Although you will also need CMake and a modern C/C++ compiler, we will omit this part
+from the tutorial as you should have them installed already by default.
+
+##### APT Packages #####
+
+To install all required APT packages at once, simply run:
+`apt-get install libboost-all-dev libpcre3-dev zlib1g-dev libgcrypt11-dev libicu-dev python libssl-dev`
+
+##### CppCMS #####
+
+To install CppCMS, there are two options available:
+
+- Download and build from source
+- Install via dependency manager
+
+To install CppCMS manually without dependency manager:
+- Download the (latest) CppCMS source from [SourceForge](https://sourceforge.net/projects/cppcms/files/cppcms/)
+- Extract the archive: `tar -xjf cppcms-1.x.x.tar.bz2 && cd cppcms-1.x.x` (replace 1.x.x with your version)
+- Configure the build: `mkdir build && cd build && cmake ..`
+- Execute the build, run tests and install: `make && make test && make install`
+
+Further build and installation information can be found on their
+[website](http://cppcms.com/wikipp/en/page/cppcms_1x_build).
+It contains also a [guide](http://cppcms.com/wikipp/en/page/apt)
+explaining the installation through a dependency manager.
+Unfortunately, no repository is available for Debian Jessie yet.
+
+##### Jansson #####
+
+The Jansson library supports working with json data in C.
+To install it, use the following steps:
+
+- Download the (latest) source from [their website](http://www.digip.org/jansson/releases/)
+- Extract the archive: `bunzip2 -c jansson-2.x.tar.bz2 | tar xf - && cd jansson-2.x` (replace 2.x with your version)
+- Configure the build: `mkdir build && cd build && cmake ..`
+- Execute the build, run tests and install: `make && make check && make install`
+
+##### LibJWT #####
+
+Now we can also install LibJWT:
+
+- Download the (latest) source from [GitHub](https://github.com/benmcollins/libjwt/releases)
+- Extract the archive: `tar -xjf v1.x.x && cd libjwt-1.x.x` (replace 1.x.x with your version)
+- Configure the build: `mkdir build && cd build && cmake ..`
+- Execute the build and run tests: `make jwt && make jwt_static && make check`
+
+Alternatively LibJWT can also be installed through a pre-built APT package:
+
+- Add APT repository: `add-apt-repository ppa:ben-collins/libjwt`
+- Install via apt-get: `apt-get update && apt-get install libjwt`
+
+#### Frontend ####
+
+The frontend does only require the package manager [npm](https://www.npmjs.com/).
+It is preferred to install it along with [nodeJS](https://nodejs.org/).
+
+To install npm via APT, use: `apt-get install npm`
+
+### Build the Applications ###
+
+After installing the dependencies, we are ready to build the applications.
+To do so, we can follow the steps explained in the [build guide](/doc/COMPILE.md).
+Make sure to include the two tools `rest-backend` and `rest-frontend`, e.g. by
+using the arguments `-DTOOLS=ALL` or `-DTOOLS=rest-backend;rest-frontend`.
+
```

Comment:
I mean you should mention it in the tutorial. A sentence and a link to doc/INSTALL.md should be enough.